### Importation des libraires nécessaires

In [1]:
import pandas as pd
import numpy as np

### Importation des bases

In [345]:
dfgender = pd.read_csv(r"C:\Users\Shokh\Desktop\Formation\ML\Projet Titanic\gender_submission.csv")
test = pd.read_csv(r"C:\Users\Shokh\Desktop\Formation\ML\Projet Titanic\test.csv")
train = pd.read_csv(r"C:\Users\Shokh\Desktop\Formation\ML\Projet Titanic\train.csv")

### Exploration du dataset

In [87]:
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Valeurs manquantes ?

In [88]:
train.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

177 valeurs manquantes dans la colonne 'Âge' (importante) et 687 dans la colonne "Cabin" (moins importante pour notre analyse)

### Description des colonnes (features) du dataset :

**PassengerID :** L'ID (unique) du passager, l'index.

**Survived :** Indique la survie ou non du passager, valeur binaire 0 (non survécu) ou 1 (survécu) type INT

**Pclass :** Classe du ticket du passager, 1ère (1), 2ème (2), 3ème (3) type INT

**Name :** Nom du passager, type STRING.

**Sex :** Sexe du passager, homme ou femme type STRING

**Age :** Âge du passager type INT

**SibSp :** Nombre de frères/soeurs ou conjoints du passager à bord

**Parch :** Nombre de parents/enfants du passager à bord

**Ticket :** Numéro du ticket

**Fare :** Prix du billet du passager

**Cabin :** Numéro de la cabine du passager

**Embarked :** Port d'embarcation du passager : C = Cherbourg, Q = Queenstown, S = Southampton

In [29]:
train.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

Pour cette commande, ici les seules colonnes pertinentes sont l'âge et le fare (prix du billet)

In [346]:
train = train.drop(columns=['Cabin','Ticket','Name'])

In [90]:
train.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch',
       'Fare', 'Embarked'],
      dtype='object')

In [91]:
train.shape

(891, 9)

In [92]:
train.head()

PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0            1         0       3    male  22.0      1      0   7.2500        S
1            2         1       1  female  38.0      1      0  71.2833        C
2            3         1       3  female  26.0      0      0   7.9250        S
3            4         1       1  female  35.0      1      0  53.1000        S
4            5         0       3    male  35.0      0      0   8.0500        S

In [4]:
train['Embarked'].value_counts()

S    644
C    168
Q     77
Name: Embarked, dtype: int64

In [347]:
train['Embarked'].fillna('S', inplace=True)

In [95]:
train.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Sex              0
Age            177
SibSp            0
Parch            0
Fare             0
Embarked         0
dtype: int64

In [348]:
train['Age'].fillna(15,limit=59,inplace=True)
train['Age'].fillna(30,limit=59,inplace=True)

In [7]:
train.isna().sum()

PassengerId     0
Survived        0
Pclass          0
Sex             0
Age            59
SibSp           0
Parch           0
Fare            0
Embarked        0
dtype: int64

In [349]:
train['Age'].fillna(45,inplace=True)

In [9]:
train.isna().sum()

PassengerId    0
Survived       0
Pclass         0
Sex            0
Age            0
SibSp          0
Parch          0
Fare           0
Embarked       0
dtype: int64

Plus aucune valeur manquantes. Il y avait 177 valeurs manquantes dans la colonne 'Age', nous avons regardé la moyenne et l'écart-type sachant que nous avons une moyenne de 30 et un écart type de 15. Ensuite nous avons divisé le nombre de valeurs manquantes par 3 pour les distribuer de façon la plus équitable possible. Donc 59 modalités par 30 (moyenne) ensuite 59 modalités par 15 (30-écart type = 15) puis 59 modalités par 45 (moyenne + écart type = 45)

In [10]:
corr_matrix = train.corr()
corr_matrix.style.background_gradient(cmap='coolwarm') # Aucune feature n'est vraiment corrélé à resilies

On observe quasiment aucune valeur corrélé avec "survived" (survie ou non) à part Pclass qui est légèrement négativement corrélé et Fare légèrement positivement corrélé, mais pas vraiment suffisant.

In [101]:
train.to_csv(r'D:\Git\Brief_Projet_Titanic\titanic\clean_train.csv')

In [102]:
train.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Sex             object
Age            float64
SibSp            int64
Parch            int64
Fare           float64
Embarked        object
dtype: object

# Regression linéaire

#### Importation librairies pour régression linéaire

In [295]:
from sklearn.linear_model import *
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
from sklearn.metrics import *
from sklearn import svm
import math

In [217]:
model = LinearRegression()

In [211]:
# Modèle regression logistique/classification
model = svm.LinearSVC()

In [13]:
train.head()

PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0            1         0       3    male  22.0      1      0   7.2500        S
1            2         1       1  female  38.0      1      0  71.2833        C
2            3         1       3  female  26.0      0      0   7.9250        S
3            4         1       1  female  35.0      1      0  53.1000        S
4            5         0       3    male  35.0      0      0   8.0500        S

### Passage du qualitative au quantitatif pour la régression linéaire

In [359]:
train.loc[train["Sex"] == "male", "Sex"] = 0
train.loc[train["Sex"] == "female", "Sex"] = 1

In [360]:
train.loc[train["Embarked"] == "S", "Embarked"] = 0
train.loc[train["Embarked"] == "C", "Embarked"] = 1
train.loc[train["Embarked"] == "Q", "Embarked"] = 2

## Prédiction avec les différentes features

Avec **l'âge** comme seule feature

In [308]:
X = train[['Age']]
y = train[['Survived']]

In [309]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [310]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [311]:
predictions = model.predict(X_test)

In [312]:
score = model.score(X_test,y_test)
print('R² = :', abs(score))

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.006032672591027799
RMSE =  0.48483607438600446


Avec pour seule feature l'âge, le score est extrêmement mauvais en régression linéaire.

Avec le **sexe** comme seule feature

In [313]:
X = train[['Sex']]
y = train[['Survived']]

In [314]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [315]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [316]:
predictions = model.predict(X_test)

In [317]:
score = model.score(X_test,y_test)
print('R² = :', score)

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.24706832287610825
RMSE =  0.40859024129928534


Avec la **classe** comme seule feature

In [318]:
X = train[['Pclass']]
y = train[['Survived']]

In [319]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [320]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [321]:
predictions = model.predict(X_test)

In [322]:
score = model.score(X_test,y_test)
print('R² = :', score)

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.07291448084358143
RMSE =  0.45773519577729926


Avec le **Fare** comme seule feature

In [323]:
X = train[['Fare']]
y = train[['Survived']]

In [324]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [325]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [326]:
predictions = model.predict(X_test)

In [327]:
score = model.score(X_test,y_test)
print('R² = :', score)

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.05870233778661271
RMSE =  0.4699730036858108


### Avec toutes les features précédentes

In [328]:
X = train[['Sex','Age','Pclass','Fare']]
y = train[['Survived']]

In [329]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [330]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [331]:
predictions = model.predict(X_test)

In [332]:
score = model.score(X_test,y_test)
print('R² = :', score)

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.3496448663774553
RMSE =  0.3822310153235341


Le meilleur score s'obtient donc quand on met toutes les features, Age Sex Class et Fare. Score bien meilleure qu'avec une seule de ces variable prises individuellement mais toujours peu convaincant, nous ne sommes qu'à 0.37.

In [333]:
X = train[['Sex','Age','Pclass']]
y = train[['Survived']]

In [334]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)

In [335]:
model.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [336]:
predictions = model.predict(X_test)

In [339]:
score = model.score(X_test,y_test)
print('R² = :', score)

mse = mean_squared_error(y, model.predict(X))
rmse = math.sqrt(mse)
print('RMSE = ', rmse)

R² = : 0.3955357337308296
RMSE =  0.3823685168866038


On peut voir que tous les score, que ça soit avec une seule feature ou toutes les features en même reste très bien quoiqu'il arrive en régression linéaire. Il n'arrive pas à prédire de façon efficace et fait moins bien que le hasard. Ceci est sûrement du au fait que la régreession linéaire n'est pas du tout adapté à ce type de problème qui est au final un problème de classfication.

On cherche à savoir si tel ou tel passager, selon plusieurs critères déterminant (ou pas) à bien plus de chance de survivre (1) ou de ne pas survivre (0), c'est donc un problème de régression logistique "simple".

Le problème vient du fait que, la régression linéaire qui est censé prédire une variable quantitative grâce à d'autres variables quantitative n'y arrive car une conversion a été faite, les données ont été transformés et ça n'est pas adapté.

# Regression logistique

In [371]:
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
from sklearn.metrics import *
import math
from sklearn.feature_selection import VarianceThreshold, SelectKBest, chi2, SelectFromModel, RFE, RFECV
from sklearn import preprocessing

In [417]:
X = train.drop('Survived', axis=1)
y = train[['Survived']]

In [429]:
scores = []
for i in range(1,101):
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=None)
    model.fit(X_train,y_train)
    predict = model.predict(X_test)
    score = model.score(X,y)
    scores.append(score)
    
final_score = np.mean(scores)
print(final_score)

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ),

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ),

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ),

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ),

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ),

0.8028507295173963


C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


In [403]:
X_scaled = preprocessing.scale(X_train)

print(selector.get_support())

selector.score(X_test,y_test)

In [410]:
selector = RFECV(LogisticRegression(),step=1,min_features_to_select=3,cv=5) # step= nb de variables eleminées par tour de boucle, 
# min_features_to_select= cooix du nb de variable min restantes
# cv= nb de split pour la cross validation
selector.fit_transform(X_train,y_train)

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number

array([[3, 0, 29.0, 0, 0, 0],
       [2, 1, 32.5, 0, 0, 0],
       [3, 0, 15.0, 0, 0, 0],
       ...,
       [2, 1, 2.0, 1, 1, 0],
       [3, 0, 45.0, 0, 0, 1],
       [3, 0, 30.5, 0, 0, 0]], dtype=object)

In [419]:
model = LogisticRegression()

In [424]:
model.fit(X_train,y_train)

C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Shokh\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [425]:
predict = model.predict(X_test)

In [426]:
model.score(X,y)

0.8013468013468014